test/System.CommandLine.StaticCompletions.Tests/VerifyExtensions.cs

```diff
@@ -13,11 +13,22 @@ public static async Task Verify(CliCommand command, IShellProvider provider, [Ca
         var completions = provider.GenerateCompletions(command);
         var settings = new VerifySettings();
         var sourceFileDir = Path.GetDirectoryName(sourceFile)!;
-        if (!Directory.Exists(sourceFileDir))
+        var snapshotDirectory = new DirectoryInfo(Path.Combine(sourceFileDir, "snapshots", provider.ArgumentName));
+        var closestExistingDirectory = GetClosestExistingDirectory(snapshotDirectory);
```

Comment:
what is snapshotDirectory resolving to in CI?

Reply:
snapshotDirectory: /mnt/vss/_work/1/s/test/System.CommandLine.StaticCompletions.Tests/snapshots/zsh
currentDirectory: /datadisks/disk1/work/B9F10A62/w/965B0881/e

I think there's something odd about the test execution setup/layout, but I don't know enough about the way the CI setup runs test code to narrow it down. Need to chat with the SDK team some more.

Reply:
@SimonCropp I talked with @MiYanni and the tests are indeed executing in a completely different environment, one where source file paths at available at compile-time are completely irrelevant.

I've done some path-munging to correctly find the snapshots in the test environment now, but the verifer is still trying to create a directory that I believe is present and explicitly passed via `settings.UseDirectory`. Is there a way to prevent any directory creation, or are there other mechanisms I can use to customize/sidestep directory location?

Reply:
Got it! https://github.com/dotnet/sdk/pull/42416/commits/823c94010323f8f9a595353fccfeee086319bd72 was what I needed.

Reply:
The key to the whole thing was getting `DerivePathInfo` overriden correctly, which (due to our build processes) meant that I needed to:

* copy snapshots to test output directories
* point the DerivePathInfo delegates to the snapshot directory off of the Environment.CurrentDirectory.

Reply:
Just pushed a change that makes all of that behavior conditional on CI, yes. Good call!